Update yolov7.py

akashAD98 opened the pull request:
i have added fps code, in which  you can see on FPS on cmd prompt as well as on the inference video,  but I'm not able to save the video & also once the video is over its not stop,i tried to add code & this logic working fine for other codes but here its failing, can you please give me suggestion

akashAD98 committed:
Update yolov7.py

i have added fps code, wichi is you can see on cmd promt as well as on inference video, 
but im not able to save the video & also once the video is over its not stopping,i tried to add code & this logic working fine for other codes but here its failling, can you please give me suggetion

akashAD98 commented:
in last frame its going to stuck & my laptop start hanging
![image](https://user-images.githubusercontent.com/62583018/220885515-e2f54ed0-bc30-4fd2-b81b-cf0cad04dcb5.png)

akashAD98 committed:
Update yolov7.py

akashAD98 commented:
updated labels with confidence scores on it. so we can easily evaluate

OpenVINO-dev-contest commented on python/yolov7.py:
```diff
@@ -240,12 +243,25 @@ def infer_cam(self, source):
+        fps = int(round(cap.get(cv2.CAP_PROP_FPS)))
+        width = int(cap.get(cv2.CAP_PROP_FRAME_WIDTH))
+        height = int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT))
+        out = cv2.VideoWriter('resultsvideo.avi',fourcc,fps,(width,height))
```
If there is a live streaming input, I think recording it as a video is unnecessary, or we should let user to make the decision, but not hard code on it.

akashAD98 replied:
thanks, ill try to fix it & also im not getting output as detected result, what should i need to pass in out.write( )

when i pass img/frame its not saving detected video

openvino-dev-samples reviewed (commented):
Hi @akashAD98  Thanks for your contribution. I believe the modules in notebooks can help to solve your problem on video processing. You can simply it and update the a new module file into this rep.
https://github.com/openvinotoolkit/openvino_notebooks/blob/main/notebooks/utils/notebook_utils.py#L207

akashAD98 commented:
solved this issue,as well as automatically once video is over it will stop video https://github.com/OpenVINO-dev-contest/YOLOv7_OpenVINO_cpp-python/issues/16

akashAD98 committed:
support for fps & automatically  video close fix 

solved issue https://github.com/OpenVINO-dev-contest/YOLOv7_OpenVINO_cpp-python/issues/16
& also once video is over ,its stucking, so that problem also solved

akashAD98 committed:
Update yolov7.py

fps & video stop code added, also 
i want to add code for saving the detection video,
out.write(frame) it's not saving detection video, actually it's saving original video only.

akashAD98 committed:
Update yolov7.py

akashAD98 committed:
Delete notebook_utils.py

akashAD98 committed:
Delete async_pipeline.py

akashAD98 committed:
Delete model.py

akashAD98 committed:
Delete custom_segmentation.py

akashAD98 committed:
Delete __init__.py

akashAD98 committed:
Add files via upload

akashAD98 committed:
Update yolov7.py

akashAD98 committed:
Update yolov7.py

akashAD98 committed:
added inference on multiple images .ie on folder

added support for multiple images, the user can pass the input folder path & output folder path & it will generate output in the output folder

example
```
python python/image.py -m model_withgrid/yolov7-vehicle40_newint8.xml -g True -i data/images/  -o data/ops/
```

akashAD98 committed:
Update image.py

akashAD98 commented:
@OpenVINO-dev-contest I hope you can accept my pr, if there is anything you want to give suggestions please let me know.

so far code working fine .1) stopping video once the last frame is over
2) fps on video & integrated openvino video player code here for assigning custom fps
3) the user do multiple image inferencing at once

OpenVINO-dev-contest commented on python/models/model.py:
```diff
@@ -0,0 +1,42 @@
+"""
```
Hi @akashAD98 My I know what's the purpose of ```model.py``` and ```custom_segmentation.py```?

akashAD98 replied:
okay thanks ill remove &unnesessory componets

akashAD98 replied:
i deleted custom_segmentation.py & kept model.py bcz we are importing pre ,post code from model file,
without that im getting no module name found error

OpenVINO-dev-contest commented on python/notebook_utils.py:
```diff
@@ -0,0 +1,702 @@
+#!/usr/bin/env python
```
I dont think you have to copy whole file from notebook to this repo, some components are unnecessary.

OpenVINO-dev-contest commented on python/async_pipeline.py:
```diff
@@ -0,0 +1,150 @@
+"""
```
what's the purpose of this file ?

akashAD98 replied:
its 

from async_pipeline import AsyncPipeline  in ,notebook_utils fil ewe imported this function

without this im getting this error

\notebook_utils.py", line 23, in <module>
    from async_pipeline import AsyncPipeline
ModuleNotFoundError: No module named 'async_pipeli

openvino-dev-samples reviewed (commented):
@akashAD98 Thanks again for your contribution, but I think you should remove some unused file or function, in order to make this repository clean and tidy.  Also you should consider the format of code: don't leave too much blank between codes.
BTW if you wish to add some new feature or function in this repository, you have update README as well, to tell our user how to play with it.

akashAD98 commented:
@OpenVINO-dev-contest yes ill make it clean & update redme

akashAD98 committed:
Update yolov7.py

akashAD98 committed:
Update image.py

akashAD98 committed:
Update README.md

doing multiple image inference at once

akashAD98 committed:
Update README.md

akashAD98 commented:
@OpenVINO-dev-contest updated with readme ,let me know if anything is missing

akashAD98 commented:
@OpenVINO-dev-contest do you accept pr? please let me know if anything is wrong here?ill improve it

OpenVINO-dev-contest commented:
Sorry for late reply, I do not have much time to do validation on it in this month, but seem some unnecessary stuff also attached within this PR, e.g ```python/models/__pycache__/__init__.cpython-38.pyc```

aiakash commented:
@OpenVINO-dev-contest okay np ,take your time
